--- CMakeLists.txt      2024-04-19 16:00:35.000000000 -0500
+++ CMakeLists.txt      2024-04-26 17:36:53.597701056 -0500
@@ -114,11 +115,7 @@
 set(ZIG_SINGLE_THREADED off CACHE BOOL "limit the zig compiler to use only 1 thread")
 set(ZIG_AR_WORKAROUND off CACHE BOOL "append 'ar' subcommand to CMAKE_AR")

-if("${ZIG_TARGET_TRIPLE}" STREQUAL "native")
-    set(ZIG_USE_LLVM_CONFIG ON CACHE BOOL "use llvm-config to find LLVM libraries")
-else()
-    set(ZIG_USE_LLVM_CONFIG OFF CACHE BOOL "use llvm-config to find LLVM libraries")
-endif()
+set(ZIG_USE_LLVM_CONFIG ON CACHE BOOL "use llvm-config to find LLVM libraries")

 if(ZIG_AR_WORKAROUND)
   string(REPLACE "<CMAKE_AR>" "<CMAKE_AR> ar" CMAKE_C_ARCHIVE_CREATE ${CMAKE_C_ARCHIVE_CREATE})
@@ -852,6 +854,21 @@
 )

 add_executable(zig2 ${ZIG2_C_SOURCE} ${ZIG_COMPILER_RT_C_SOURCE})
+add_custom_command(
+  TARGET zig2 POST_BUILD
+  COMMAND patchelf
+      --set-interpreter ${LIBC_INTERPRETER}
+      --replace-needed libc.so libc-${LIBC_CONDA_VERSION}.so
+      --add-needed libc-${LIBC_CONDA_VERSION}.so
+      --add-needed libm-${LIBC_CONDA_VERSION}.so
+      --add-needed libpthread-${LIBC_CONDA_VERSION}.so
+      --add-needed librt-${LIBC_CONDA_VERSION}.so
+      --add-needed libdl-${LIBC_CONDA_VERSION}.so
+      --set-rpath "${LIBC_RPATH}"
+      zig2 && ldd zig2 > _log_zig2.txt
+  DEPENDS zig2
+  COMMENT STATUS "[---%] Patching zig2 with LIBC ${LIBC_CONDA_VERSION}"
+)
 set_target_properties(zig2 PROPERTIES
   COMPILE_FLAGS ${ZIG2_COMPILE_FLAGS}
   LINK_FLAGS "${ZIG2_LINK_FLAGS}"
@@ -934,8 +945,20 @@

 add_custom_command(
   OUTPUT "${CMAKE_BINARY_DIR}/stage3/bin/zig"
-  COMMAND zig2 build --prefix "${CMAKE_BINARY_DIR}/stage3" ${ZIG_BUILD_ARGS}
-  COMMENT STATUS "Building stage3"
+  COMMAND zig2 build --prefix "${CMAKE_BINARY_DIR}/stage3" ${ZIG_BUILD_ARGS} &&
+          ldd "${CMAKE_BINARY_DIR}/stage3/bin/zig" > _log_zig.txt &&
+          patchelf
+              --set-interpreter ${LIBC_INTERPRETER}
+              --replace-needed libc.so libc-${LIBC_CONDA_VERSION}.so
+              --add-needed libc-${LIBC_CONDA_VERSION}.so
+              --add-needed libm-${LIBC_CONDA_VERSION}.so
+              --add-needed libpthread-${LIBC_CONDA_VERSION}.so
+              --add-needed librt-${LIBC_CONDA_VERSION}.so
+              --add-needed libdl-${LIBC_CONDA_VERSION}.so
+              --set-rpath "${LIBC_RPATH}"
+              "${CMAKE_BINARY_DIR}/stage3/bin/zig" &&
+          ldd "${CMAKE_BINARY_DIR}/stage3/bin/zig" > _log2_zig.txt
+  COMMENT STATUS "Building stage3 & Patching stage3/bin/zig with LIBC ${LIBC_CONDA_VERSION}"
   WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
 )

