--- powerpc64le-conda-linux-gnu/sysroot/usr/include/signal.h       2024-06-19 11:12:39.145132711 -0500
+++ powerpc64le-conda-linux-gnu/sysroot/usr/include/signal.h       2024-06-19 11:12:31.340976338 -0500
@@ -237,8 +237,8 @@
 extern int sigsuspend (const sigset_t *__set) __nonnull ((1));

 /* Get and/or set the action for signal SIG.  */
-extern int sigaction (int __sig, const struct sigaction *__restrict __act,
-              struct sigaction *__restrict __oact) __THROW;
+/* extern int sigaction (int __sig, const struct sigaction *__restrict __act,
+              struct sigaction *__restrict __oact) __THROW; */

 /* Put in SET all signals that are blocked and waiting to be delivered.  */
 extern int sigpending (sigset_t *__set) __THROW __nonnull ((1));
