--- CMakeLists.txt.old	2024-06-06 14:05:11.000000000 -0500
+++ CMakeLists.txt	2024-06-22 11:27:13.931633400 -0500
@@ -164,2 +164,4 @@
 endif()
+string(REPLACE ".dll" "" LLVM_LIBRARIES "${LLVM_LIBRARIES}")
+set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")  #list(APPEND LLVM_LIBRARIES "c")

@@ -731,2 +733,3 @@
 else()
+target_compile_options(zigcpp PRIVATE -MD)
 target_compile_options(zigcpp PRIVATE /Zc:preprocessor)
@@ -901,3 +904,3 @@
   COMPILE_FLAGS ${ZIG2_COMPILE_FLAGS}
-  LINK_FLAGS "${ZIG2_LINK_FLAGS}"
+  LINK_FLAGS "${ZIG2_LINK_FLAGS}" /NODEFAULTLIB:LIBCMT
 )
