--- cmake/cxx_flag_overrides.cmake      2024-04-19 16:00:35.000000000 -0500
+++ cmake/cxx_flag_overrides.cmake      2024-06-16 17:27:28.017267908 -0500
@@ -3,11 +3,11 @@
         "/D_DEBUG /MTd /Zi /Ob0 /Od /RTC1")

     set(CMAKE_CXX_FLAGS_MINSIZEREL_INIT
-        "/MT /O1 /Ob1 /D NDEBUG")
+        "/MD /O1 /Ob1 /D NDEBUG")

     set(CMAKE_CXX_FLAGS_RELEASE_INIT
-        "/MT /O2 /Ob1 /D NDEBUG")
+        "/MD /O2 /Ob1 /D NDEBUG")

     set(CMAKE_CXX_FLAGS_RELWITHDEBINFO_INIT
-        "/MT /Zi /O2 /Ob1 /D NDEBUG")
+        "/MD /Zi /O2 /Ob1 /D NDEBUG")
 endif()
