diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef371fc..01a4f23 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -915,6 +915,7 @@ set(ZIG_BUILD_ARGS
   "-Dconfig_h=${ZIG_CONFIG_H_OUT}"
 
   -Dno-langref
+  --maxrss 7000000000
 )
 
 set(ZIG_EXTRA_BUILD_ARGS "" CACHE STRING "Extra zig build args")
diff --git a/build.zig b/build.zig
index 37523e6..a151dc6 100644
--- a/build.zig
+++ b/build.zig
@@ -738,7 +738,7 @@ fn addCompilerMod(b: *std.Build, options: AddCompilerModOptions) *std.Build.Modu
 fn addCompilerStep(b: *std.Build, options: AddCompilerModOptions) *std.Build.Step.Compile {
     const exe = b.addExecutable(.{
         .name = "zig",
-        .max_rss = 7_800_000_000,
+        .max_rss = 7_000_000_000,
         .root_module = addCompilerMod(b, options),
     });
     exe.stack_size = stack_size;
