--- CMakeLists.txt.orig	2024-06-06 14:05:11.000000000 -0500
+++ CMakeLists.txt	2024-06-22 11:27:13.931633400 -0500
@@ -915,6 +915,7 @@
   "-Dconfig_h=${ZIG_CONFIG_H_OUT}"

   -Dno-langref
+  --maxrss 7000000000
 )

 set(ZIG_EXTRA_BUILD_ARGS "" CACHE STRING "Extra zig build args")
--- build.zig.orig	2024-06-06 14:05:11.000000000 -0500
+++ build.zig	2024-06-22 11:27:13.931633400 -0500
@@ -738,7 +738,7 @@
 fn addCompilerStep(b: *std.Build, options: AddCompilerModOptions) *std.Build.Step.Compile {
     const exe = b.addExecutable(.{
         .name = "zig",
-        .max_rss = 7_800_000_000,
+        .max_rss = 7_000_000_000,
         .root_module = addCompilerMod(b, options),
     });
     exe.stack_size = stack_size;
