--- CMakeLists.txt.old	2024-06-06 14:05:11.000000000 -0500
+++ CMakeLists.txt	2024-06-22 11:27:13.931633400 -0500
@@ -164,2 +164,4 @@
 endif()
+string(REPLACE ".dll" "" LLVM_LIBRARIES "${LLVM_LIBRARIES}")
+list(APPEND LLVM_LIBRARIES "c")

@@ -731,2 +733,3 @@
 else()
+set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")  #target_compile_options(zigcpp PRIVATE -MD)
 target_compile_options(zigcpp PRIVATE /Zc:preprocessor)
