--- doc/langref.html.in	2024-05-22 15:02:31.000000000 -0500
+++ doc/langref.html.in	2024-05-26 13:15:20.077402496 -0500
@@ -6038,7 +6038,6 @@
       {#header_close#}
       {#header_open|Attempt to Unwrap Null#}
       <p>At compile-time:</p>
-      {#code|test_comptime_unwrap_null.zig#}

       <p>At runtime:</p>
       {#code|runtime_unwrap_null.zig#}
@@ -6461,7 +6460,6 @@
       The {#syntax#}@cImport{#endsyntax#} builtin function can be used
       to directly import symbols from <code class="file">.h</code> files:
       </p>
-      {#code|cImport_builtin.zig#}

       <p>
       The {#syntax#}@cImport{#endsyntax#} function takes an expression as a parameter.
@@ -6570,7 +6568,6 @@
         To see where the cached files are stored when compiling code that uses {#syntax#}@cImport{#endsyntax#},
         use the <kbd>--verbose-cimport</kbd> flag:
       </p>
-      {#code|verbose_cimport_flag.zig#}

       <p>
         <code class="file">cimport.h</code> contains the file to translate (constructed from calls to
@@ -6673,7 +6670,6 @@

       {#header_open|C Variadic Functions#}
       <p>Zig supports extern variadic functions.</p>
-      {#code|test_variadic_function.zig#}

       <p>
         Variadic functions can be implemented using {#link|@cVaStart#}, {#link|@cVaEnd#}, {#link|@cVaArg#} and {#link|@cVaCopy#}.
