--- CMakeLists.txt.old	2024-06-06 14:05:11.000000000 -0500
+++ CMakeLists.txt	2024-06-22 11:27:13.931633400 -0500
@@ -163,2 +163,8 @@
     list(APPEND LLVM_LIBRARIES "${ZSTD}")
+else()
+    if (MSVC)
+        list(APPEND LLVM_LIBRARIES zstd)
+        list(APPEND LLVM_LIBRARIES xml2)
+        list(APPEND LLVM_LIBRARIES zlib)
+    endif()
 endif()
@@ -731,2 +737,3 @@
 else()
+target_compile_options(zigcpp PRIVATE -MD)
 target_compile_options(zigcpp PRIVATE /Zc:preprocessor)
@@ -808,2 +815,13 @@
 set(ZIG_HOST_TARGET_TRIPLE "${ZIG_HOST_TARGET_ARCH}-${ZIG_HOST_TARGET_OS}${ZIG_HOST_TARGET_ABI}" CACHE STRING "Host zig target triple.")
+if(DEFINED ENV{CROSSCOMPILING_EMULATOR})
+  set(CROSSCOMPILING_EMULATOR $ENV{CROSSCOMPILING_EMULATOR})
+  if(DEFINED ENV{ZIG_CROSS_TARGET_TRIPLE})
+    set(ZIG_CROSS_TARGET_TRIPLE $ENV{ZIG_CROSS_TARGET_TRIPLE} CACHE STRING "Host zig target triple.")
+    message(STATUS "Setting ZIG_CROSS_TARGET_TRIPLE ${ZIG_CROSS_TARGET_TRIPLE}")
+  else()
+    message(SEND_ERROR "ZIG_CROSS_TARGET_TRIPLE is not set while cross-compiling")
+  endif()
+else()
+  set(CROSSCOMPILING_EMULATOR "")
+endif()
 
@@ -818,2 +836,5 @@
   set(ZIG2_COMPILE_FLAGS "-std=c99 -O0 -fno-stack-protector")
+  if(ZIG_HOST_TARGET_ARCH MATCHES "^powerpc")
+    list(APPEND ZIG2_COMPILE_FLAGS "--save-temps -mlongcall")
+  endif()
   if(APPLE)
@@ -840,3 +861,3 @@
   OUTPUT "${ZIG1_C_SOURCE}"
-  COMMAND zig-wasm2c "${ZIG1_WASM_MODULE}" "${ZIG1_C_SOURCE}"
+  COMMAND ${CROSSCOMPILING_EMULATOR} ${PROJECT_BINARY_DIR}/zig-wasm2c "${ZIG1_WASM_MODULE}" "${ZIG1_C_SOURCE}"
   DEPENDS zig-wasm2c "${ZIG1_WASM_MODULE}"
@@ -852,3 +873,9 @@
 else()
-  target_link_libraries(zig1 LINK_PUBLIC m)
+  if(WIN32)
+    find_library(M NAMES m libm.lib NAMES_PER_DIR)
+    target_link_libraries(zig1 LINK_PUBLIC $M)
+    # target_link_libraries(zig1 LINK_PUBLIC $ENV{PREFIX}/Library/lib/libm.lib)
+  else()
+    target_link_libraries(zig1 LINK_PUBLIC m)
+  endif()
   if(MINGW)
@@ -858,2 +885,5 @@
 
+if(NOT DEFINED ZIG_CROSS_TARGET_TRIPLE)
+    set(ZIG_CROSS_TARGET_TRIPLE ${ZIG_HOST_TARGET_TRIPLE})
+endif()
 set(BUILD_ZIG2_ARGS
@@ -863,3 +893,3 @@
   -femit-bin="${ZIG2_C_SOURCE}"
-  -target "${ZIG_HOST_TARGET_TRIPLE}"
+  -target "${ZIG_CROSS_TARGET_TRIPLE}"
   --dep "build_options"
@@ -873,3 +903,3 @@
   OUTPUT "${ZIG2_C_SOURCE}"
-  COMMAND zig1 ${BUILD_ZIG2_ARGS}
+  COMMAND ${CROSSCOMPILING_EMULATOR} ${PROJECT_BINARY_DIR}/zig1 ${BUILD_ZIG2_ARGS}
   DEPENDS zig1 "${ZIG_STAGE2_SOURCES}"
@@ -884,3 +914,3 @@
   -femit-bin="${ZIG_COMPILER_RT_C_SOURCE}"
-  -target "${ZIG_HOST_TARGET_TRIPLE}"
+  -target "${ZIG_CROSS_TARGET_TRIPLE}"
   --dep "build_options"
@@ -892,3 +922,3 @@
   OUTPUT "${ZIG_COMPILER_RT_C_SOURCE}"
-  COMMAND zig1 ${BUILD_COMPILER_RT_ARGS}
+  COMMAND ${CROSSCOMPILING_EMULATOR} ${PROJECT_BINARY_DIR}/zig1 ${BUILD_COMPILER_RT_ARGS}
   DEPENDS zig1 "${ZIG_STAGE2_SOURCES}"
@@ -978,3 +1008,4 @@
   OUTPUT "${PROJECT_BINARY_DIR}/stage3/bin/zig"
-  COMMAND zig2 build --prefix "${PROJECT_BINARY_DIR}/stage3" ${ZIG_BUILD_ARGS}
+  COMMAND ${CROSSCOMPILING_EMULATOR} ${PROJECT_BINARY_DIR}/zig2 build --prefix "${PROJECT_BINARY_DIR}/stage3" ${ZIG_BUILD_ARGS}
+  DEPENDS zig2
   COMMENT STATUS "Building stage3"
