--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -147,6 +147,8 @@
     find_library(ZLIB NAMES libz.a libzlibstatic.a z zlib libz NAMES_PER_DIR)
     list(APPEND LLVM_LIBRARIES "${ZLIB}")
 endif()
+string(REPLACE ".dll" "" LLVM_LIBRARIES "${LLVM_LIBRARIES}")
+list(APPEND LLVM_LIBRARIES "xml2.lib" "zstd.lib" "zlib.lib" "msvcrt.lib")

 if(ZIG_STATIC_ZSTD)
     if (MSVC)
@@ -708,6 +710,8 @@
     -Wno-comment
   )
 else()
+target_compile_options(zigcpp PRIVATE -MD)
+add_compile_options(/MD /wd4715 /wd4291)
 target_compile_options(zigcpp PRIVATE /Zc:preprocessor)
 set_property(TARGET zigcpp PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded")
 endif()
@@ -828,7 +832,7 @@
 if(MSVC)
   target_link_options(zig1 PRIVATE /STACK:0x10000000)
 else()
-  target_link_libraries(zig1 LINK_PUBLIC m)
+  # target_link_libraries(zig1 LINK_PUBLIC m)
   if(MINGW)
     target_link_options(zig1 PRIVATE -Wl,--stack,0x10000000)
   endif()
@@ -915,6 +919,7 @@
   "-Dconfig_h=${ZIG_CONFIG_H_OUT}"

   -Dno-langref
+  --search-prefix "$ENV{PREFIX}/Library/lib"
   --maxrss 7000000000
 )

