--- CMakeLists.txt.old	2024-06-06 14:05:11.000000000 -0500
+++ CMakeLists.txt	2024-06-22 11:27:13.931633400 -0500
@@ -148,2 +148,5 @@
 endif()
+string(REPLACE ".dll" "" LLVM_LIBRARIES "${LLVM_LIBRARIES}")
+list(APPEND LLVM_LIBRARIES "xml2.lib" "zstd.lib" "zlib.lib") # list(APPEND LLVM_LIBRARIES "xml2.lib" "zstd" "zlib")
+# list(REMOVE_ITEM CLANG_LIBRARIES "version.lib")

@@ -725,2 +728,4 @@
 else()
+target_compile_options(zigcpp PRIVATE -MD)
+add_compile_options(/MD /wd4715 /wd4291)
 target_compile_options(zigcpp PRIVATE /Zc:preprocessor)
@@ -847,1 +850,1 @@
-  target_link_libraries(zig1 LINK_PUBLIC m)
+  # target_link_libraries(zig1 LINK_PUBLIC m)
@@ -933,2 +937,6 @@
   -Dno-langref
+  --verbose-link
+  --libc "$ENV{RECIPE_DIR}/patches/win_libc.txt"
+  --search-prefix "C:/Program Files (x86)/Windows Kits/10/Lib/10.0.22621.0/um/x64"
+  --search-prefix "C:/Windows/System32"
   --maxrss 7000000000
@@ -949 +1014 @@
-if(ZIG_STATIC AND NOT MSVC)
+if(0)
