--- a/src/codegen/llvm.zig
+++ b/src/codegen/llvm.zig
@@ -295,13 +295,8 @@
         .semver,
         .windows,
         => {},
-        inline .hurd, .linux => |ver| if (target.abi.isGnu()) {
-            try llvm_triple.print("{d}.{d}.{d}", .{
-                ver.glibc.major,
-                ver.glibc.minor,
-                ver.glibc.patch,
-            });
-        } else if (@TypeOf(ver) == std.Target.Os.LinuxVersionRange and target.abi.isAndroid()) {
+        .hurd => {},
+        .linux => |ver| if (target.abi.isAndroid()) {
             try llvm_triple.print("{d}", .{ver.android});
         },
     }
