--- CMakeLists.txt.old	2024-06-06 14:05:11.000000000 -0500
+++ CMakeLists.txt	2024-06-22 11:27:13.931633400 -0500
@@ -813,3 +817,3 @@
   set(ZIG2_COMPILE_FLAGS "/Od")
-  set(ZIG2_LINK_FLAGS "/STACK:16777216 /FORCE:MULTIPLE")
+  set(ZIG2_LINK_FLAGS "/STACK:16777216 /FORCE:MULTIPLE")
   set(ZIG2_LINK_FLAGS_DEBUG "/DEBUG")
 else()
@@ -852,3 +854,3 @@
 else()
-  target_link_libraries(zig1 LINK_PUBLIC m)
+  # target_link_libraries(zig1 LINK_PUBLIC m)
   if(MINGW)
