--- src/link/Lld.zig.orig	2024-06-06 14:05:11.000000000 -0500
+++ src/link/Lld.zig	2024-06-22 11:27:13.931633400 -0500
@@ -684,7 +684,7 @@
                             .dynamic => "",
                         };
                         try argv.append(try allocPrint(arena, "{s}vcruntime.lib", .{lib_str}));
-                        try argv.append(try allocPrint(arena, "{s}ucrt.lib", .{lib_str}));
+                        // try argv.append(try allocPrint(arena, "{s}ucrt.lib", .{lib_str}));

                         //Visual C++ 2015 Conformance Changes
                         //https://msdn.microsoft.com/en-us/library/bb531344.aspx
