--- a/src/mingw.zig.old	2024-06-06 14:05:11.000000000 -0500
+++ b/src/mingw.zig	2024-06-22 11:27:13.931633400 -0500
@@ -1018,15 +1018,15 @@
-    "api-ms-win-crt-conio-l1-1-0",
-    "api-ms-win-crt-convert-l1-1-0",
-    "api-ms-win-crt-environment-l1-1-0",
-    "api-ms-win-crt-filesystem-l1-1-0",
-    "api-ms-win-crt-heap-l1-1-0",
-    "api-ms-win-crt-locale-l1-1-0",
-    "api-ms-win-crt-math-l1-1-0",
-    "api-ms-win-crt-multibyte-l1-1-0",
-    "api-ms-win-crt-private-l1-1-0",
-    "api-ms-win-crt-process-l1-1-0",
-    "api-ms-win-crt-runtime-l1-1-0",
-    "api-ms-win-crt-stdio-l1-1-0",
-    "api-ms-win-crt-string-l1-1-0",
-    "api-ms-win-crt-time-l1-1-0",
-    "api-ms-win-crt-utility-l1-1-0",
+    // "api-ms-win-crt-conio-l1-1-0",
+    // "api-ms-win-crt-convert-l1-1-0",
+    // "api-ms-win-crt-environment-l1-1-0",
+    // "api-ms-win-crt-filesystem-l1-1-0",
+    // "api-ms-win-crt-heap-l1-1-0",
+    // "api-ms-win-crt-locale-l1-1-0",
+    // "api-ms-win-crt-math-l1-1-0",
+    // "api-ms-win-crt-multibyte-l1-1-0",
+    // "api-ms-win-crt-private-l1-1-0",
+    // "api-ms-win-crt-process-l1-1-0",
+    // "api-ms-win-crt-runtime-l1-1-0",
+    // "api-ms-win-crt-stdio-l1-1-0",
+    // "api-ms-win-crt-string-l1-1-0",
+    // "api-ms-win-crt-time-l1-1-0",
+    // "api-ms-win-crt-utility-l1-1-0",
--- a/src/link/Coff.zig.old	2024-06-06 14:05:11.000000000 -0500
+++ b/src/link/Coff.zig	2024-06-22 11:27:13.931633400 -0500
@@ -2123 +2123 @@
-                        try argv.append(try allocPrint(arena, "{s}ucrt{s}.lib", .{ lib_str, d_str }));
+                        try argv.append(try allocPrint(arena, "{s}ucrt{s}.lib", .{ lib_str, d_str }));
