--- build.zig.old	2024-06-06 14:05:11.000000000 -0500
+++ build.zig	2024-06-22 11:27:13.931633400 -0500
@@ -686,1 +686,2 @@
     exe.addLibraryPath(.{ .cwd_relative = cfg.llvm_lib_dir });
+    exe.addLibraryPath(.{ .cwd_relative = cfg.system_lib_dir });
@@ -859,1 +860,2 @@
     llvm_lib_dir: []const u8,
+    system_lib_dir: []const u8,
@@ -926,1 +928,2 @@
         .llvm_lib_dir = undefined,
+        .system_lib_dir = undefined,
@@ -985,4 +987,8 @@
             .field = "llvm_lib_dir",
         },
+        .{
+            .prefix = "#define ZIG_SYSTEM_LIB_PATH ",
+            .field = "system_lib_dir",
+        },
        .{
            .prefix = "#define ZIG_SYSTEM_LIBCXX",
