--- a/src/codegen/llvm.zig
+++ b/src/codegen/llvm.zig
@@ -1609,7 +1609,7 @@
             const lf = comp.bin_file orelse break :coff_export_flags;
             const lld = lf.cast(.lld) orelse break :coff_export_flags;
             const coff = switch (lld.ofmt) {
-                .elf, .wasm => break :coff_export_flags,
+                .elf, .wasm, .macho => break :coff_export_flags,
                 .coff => |*coff| coff,
             };
             if (!ip.isFunctionType(ip.getNav(nav_index).typeOf(ip))) break :coff_export_flags;
