--- a/src/link/Elf/relocation.zig	2026-04-13 20:19:17
+++ b/src/link/Elf/relocation.zig	2026-04-24 10:12:58
@@ -73,11 +73,25 @@
     .{ .tlsdesc, .TLSDESC },
 });
 
+const powerpc64_relocs = Table(10, elf.R_PPC64, .{
+    .{ .none, .NONE },
+    .{ .abs, .ADDR64 },
+    .{ .copy, .COPY },
+    .{ .rel, .RELATIVE },
+    .{ .irel, .IRELATIVE },
+    .{ .glob_dat, .GLOB_DAT },
+    .{ .jump_slot, .JMP_SLOT },
+    .{ .dtpmod, .DTPMOD64 },
+    .{ .dtpoff, .DTPREL64 },
+    .{ .tpoff, .TPREL64 },
+});
+
 pub fn decode(r_type: u32, cpu_arch: std.Target.Cpu.Arch) ?Kind {
     return switch (cpu_arch) {
         .x86_64 => x86_64_relocs.decode(r_type),
         .aarch64, .aarch64_be => aarch64_relocs.decode(r_type),
         .riscv64, .riscv64be => riscv64_relocs.decode(r_type),
+        .powerpc64, .powerpc64le => powerpc64_relocs.decode(r_type),
         else => @panic("TODO unhandled cpu arch"),
     };
 }
@@ -87,6 +101,7 @@
         .x86_64 => x86_64_relocs.encode(kind),
         .aarch64, .aarch64_be => aarch64_relocs.encode(kind),
         .riscv64, .riscv64be => riscv64_relocs.encode(kind),
+        .powerpc64, .powerpc64le => powerpc64_relocs.encode(kind),
         else => @panic("TODO unhandled cpu arch"),
     };
 }
@@ -106,6 +121,10 @@
                 .@"32" => .@"32",
                 .@"64" => .@"64",
             })),
+            .powerpc64, .powerpc64le => @intFromEnum(@as(elf.R_PPC64, switch (format) {
+                .@"32" => .ADDR32,
+                .@"64" => .ADDR64,
+            })),
             else => @panic("TODO unhandled cpu arch"),
         };
     }
@@ -141,6 +160,14 @@
                 },
                 .debug_frame => unreachable,
             })),
+            .powerpc64, .powerpc64le => @intFromEnum(@as(elf.R_PPC64, switch (source_section) {
+                else => switch (address_size) {
+                    .@"32" => .ADDR32,
+                    .@"64" => .ADDR64,
+                    else => unreachable,
+                },
+                .debug_frame => unreachable,
+            })),
             else => @panic("TODO unhandled cpu arch"),
         };
     }
@@ -166,6 +193,7 @@
         .x86_64 => try writer.print("R_X86_64_{s}", .{@tagName(@as(elf.R_X86_64, @enumFromInt(r_type)))}),
         .aarch64, .aarch64_be => try writer.print("R_AARCH64_{s}", .{@tagName(@as(elf.R_AARCH64, @enumFromInt(r_type)))}),
         .riscv64, .riscv64be => try writer.print("R_RISCV_{s}", .{@tagName(@as(elf.R_RISCV, @enumFromInt(r_type)))}),
+        .powerpc64, .powerpc64le => try writer.print("R_PPC64_{s}", .{@tagName(@as(elf.R_PPC64, @enumFromInt(r_type)))}),
         else => unreachable,
     }
 }
